src/bespokelabs/curator/request_processor/litellm_online_request_processor.py

```diff
@@ -0,0 +1,301 @@
+        top_p (Optional[float]): Top-p sampling parameter
+        presence_penalty (Optional[float]): Presence penalty for response diversity
+        frequency_penalty (Optional[float]): Frequency penalty for response diversity
+        client: Instructor-wrapped LiteLLM client for structured outputs
+    """
+
+    def __init__(
+        self,
+        model: str,
+        temperature: Optional[float] = None,
+        top_p: Optional[float] = None,
+        presence_penalty: Optional[float] = None,
+        frequency_penalty: Optional[float] = None,
+    ):
+        super().__init__(
+            model=model,
+            temperature=temperature,
+            top_p=top_p,
+            presence_penalty=presence_penalty,
+            frequency_penalty=frequency_penalty,
+        )
+        self.client = instructor.from_litellm(litellm.acompletion)
+
+    def check_structured_output_support(self, prompt_formatter: PromptFormatter):
+        """Verify if the model supports structured output via instructor.
+
+        Tests the model's capability to handle structured output by making a test request
+        with a simple schema.
+
+        Args:
+            prompt_formatter (PromptFormatter): Contains response format requirements
+
+        Returns:
+            bool: True if structured output is supported, False otherwise
+
+        Note:
+            - Uses a simple User schema as test case
+            - Logs detailed information about support status
+            - Required for models that will use JSON schema responses
+        """
+        assert prompt_formatter.response_format
+
+        class User(BaseModel):
+            name: str
+            age: int
+
+        try:
+            client = instructor.from_litellm(litellm.completion)
+            response = client.chat.completions.create(
+                model=self.model,
+                messages=[{"role": "user", "content": "Jason is 25 years old."}],
+                response_model=User,
+            )
+            logger.info(f"Check instructor structure output response: {response}")
+            assert isinstance(response, User)
+            logger.info(
+                f"Model {self.model} supports structured output via instructor, response: {response}"
+            )
+            return True
+        except Exception as e:
+            logger.warning(
+                f"Model {self.model} does not support structured output via instructor: {e}"
+            )
+            return False
+
+    def estimate_output_tokens(self) -> int:
+        """Estimate the number of tokens in the model's response.
+
+        Uses LiteLLM's get_max_tokens and applies a conservative estimate
+        by dividing by 4 to avoid hitting context limits.
+
+        Returns:
+            int: Estimated number of output tokens
+
+        Note:
+            Falls back to 0 if token estimation fails
+        """
+        try:
+            return litellm.get_max_tokens(model=self.model) // 4
```

Comment:
hard to do this well..... different models have different output lengths. Most conservative would be max tokens. But // 4 will vary per model.